tools/ci-build/changelogger/src/render.rs

```diff
@@ -592,9 +676,69 @@ author = "rcoh"
     #[test]
     fn test_empty_render() {
         let smithy_rs = Vec::<ChangelogEntry>::new();
-        let (release_title, release_notes) = render(&smithy_rs, "some header");
+        let (release_title, release_notes) =
+            render(&smithy_rs, CrateVersionMetadataMap::new(), "some header");
 
         assert_eq!(release_title, "some header\n===========\n");
         assert_eq!(release_notes, "");
     }
+
+    #[test]
+    fn test_crate_versions() {
+        let mut crate_version_metadata_map = CrateVersionMetadataMap::new();
+        crate_version_metadata_map.insert(
+            "aws-config".to_owned(),
+            CrateVersion {
+                category: PackageCategory::AwsRuntime,
+                version: "0.54.1".to_owned(),
+                source_hash: "e93380cfbd05e68d39801cbf0113737ede552a5eceb28f4c34b090048d539df9"
+                    .to_owned(),
+                model_hash: None,
+            },
+        );
+        crate_version_metadata_map.insert(
+            "aws-sdk-accessanalyzer".to_owned(),
+            CrateVersion {
+                category: PackageCategory::AwsSdk,
+                version: "0.24.0".to_owned(),
+                source_hash: "a7728756b41b33d02f68a5865d3456802b7bc3949ec089790bc4e726c0de8539"
+                    .to_owned(),
+                model_hash: Some(
+                    "71f1f130504ebd55396c3166d9441513f97e49b281a5dd420fd7e2429860b41b".to_owned(),
+                ),
+            },
+        );
+        crate_version_metadata_map.insert(
+            "aws-smithy-async".to_owned(),
+            CrateVersion {
+                category: PackageCategory::SmithyRuntime,
+                version: "0.54.1".to_owned(),
+                source_hash: "8ced52afc783cbb0df47ee8b55260b98e9febdc95edd796ed14c43db5199b0a9"
+                    .to_owned(),
+                model_hash: None,
+            },
+        );
+        let (release_title, release_notes) = render(
+            &Vec::<ChangelogEntry>::new(),
+            crate_version_metadata_map,
+            "some header",
+        );
+
+        assert_eq!(release_title, "some header\n===========\n");
+        let expected_body = r#"
+**Crate Versions**
+<details>
+<summary>Click to expand to view crate versions...</summary>
+
+Crate|Version|
+-|-|
+aws-config|0.54.1|
+aws-sdk-accessanalyzer|0.24.0|
+aws-smithy-async|0.54.1|
```

Comment:
IIRC the recommended formatting is to have a pipe on both ends of the row. I don't think it's strictly necessary though, so feel free to ignore it.

Reply:
Thanks for calling it out. I prefer having a pipe on both ends for consistency. Addressed in 7d5524d.